File: backend/app/schemas/file_analysis.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DownloadFileResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    download_task_id: int
    file_index: int
    provider_file_id: str | None
    parent_id: str | None
    name: str
    size: int
    progress: float
    priority: int
    file_type: str
    selected: bool
    analysis_score: int
    season_number: int | None
    episode_number: int | None
    created_at: datetime
    updated_at: datetime


class RenamePreviewResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    download_file_id: int
    task_id: int | None
    file_id: str | None
    parent_id: str | None
    original_name: str
    target_name: str
    original_path: str
    target_path: str
    file_size: int
    file_type: str
    episode_number: int | None
    confidence: int
    conflict: bool
    warning_message: str | None
    status: str
    created_at: datetime
    updated_at: datetime


class DownloadFileListResponse(BaseModel):
    message: str
    files: list[DownloadFileResponse]


class DownloadFileMessageResponse(BaseModel):
    message: str
    file: DownloadFileResponse


class DownloadFileUpdateRequest(BaseModel):
    selected: bool | None = None
    file_type: str | None = Field(
        default=None,
        pattern="^(video|subtitle|image|sample|document|other|unknown)$",
    )
    season_number: int | None = Field(default=None, ge=0)
    episode_number: int | None = Field(default=None, ge=1)


class FileAnalysisMessageResponse(BaseModel):
    message: str
    files: list[DownloadFileResponse]


class RenamePreviewListResponse(BaseModel):
    message: str
    previews: list[RenamePreviewResponse]


class SimpleMessageResponse(BaseModel):
    message: str

File: frontend/src/types/fileAnalysis.ts
export type DownloadFileType = 'video' | 'subtitle' | 'image' | 'sample' | 'document' | 'other' | 'unknown'

export interface DownloadFile {
  id: number
  download_task_id: number
  file_index: number
  provider_file_id: string | null
  parent_id: string | null
  name: string
  size: number
  progress: number
  priority: number
  file_type: DownloadFileType
  selected: boolean
  analysis_score: number
  season_number: number | null
  episode_number: number | null
  created_at: string
  updated_at: string
}

export interface RenamePreview {
  id: number
  download_file_id: number
  task_id: number | null
  file_id: string | null
  parent_id: string | null
  original_name: string
  target_name: string
  original_path: string
  target_path: string
  file_size: number
  file_type: DownloadFileType
  episode_number: number | null
  confidence: number
  conflict: boolean
  warning_message: string | null
  status: string
  created_at: string
  updated_at: string
}

export interface DownloadFileListResponse {
  message: string
  files: DownloadFile[]
}

export interface DownloadFileMessageResponse {
  message: string
  file: DownloadFile
}

export interface DownloadFileUpdateRequest {
  selected?: boolean
  file_type?: DownloadFileType
  season_number?: number | null
  episode_number?: number | null
}

export interface FileAnalysisMessageResponse {
  message: string
  files: DownloadFile[]
}

export interface RenamePreviewListResponse {
  message: string
  previews: RenamePreview[]
}

export interface SimpleMessageResponse {
  message: string
}
